lib/chef/resource/aws_cloudwatch_alarm.rb

```diff
@@ -0,0 +1,28 @@
+require 'chef/provisioning/aws_driver/aws_resource'
+
+class Chef::Resource::AwsCloudwatchAlarm < Chef::Provisioning::AWSDriver::AWSResource
+  include Chef::Provisioning::AWSDriver::AWSTaggable
+
+  aws_sdk_type AWS::CloudWatch::Alarm, id: :name
+
+  attribute :name, :kind_of => String, :name_attribute => true
```

Comment:
Would be nice, since this is a new provider, to respect the new "standard" of RFC 054:
https://github.com/chef/chef-rfc/blob/master/rfc054-resource-attribute-improvements.md